### PR TITLE
WIP: Start migrating away from the legacy Raspberry Pi stack

### PR DESCRIPTION
**Pull request by chewi**
This mentions libcamera in the commit messages, but we don't actually that yet. Sam is working on it.

### CODE REVIEW

**Comment by chewi** [comment]
I've now tested this in 32-bit and 64-bit with KMS and FKMS. H.265/HEVC videos are still struggling on my Pi 3, but I understand that this model doesn't have the hardware for this. John's patch set mentions a special rpi_hevc decoder that works even on a Pi 3, but it looks like this may have been an old experiment that only lived in other branches. I want to touch base with John anyway, so I'll ask him about it.

**Comment by chewi** [comment]
Just tried my Pi 4. First time I'd powered it on! Kodi was fine, but mpv was worse than ever. Very odd. I'll speak to John.

**Comment by chewi** [comment]
Figured it out. It wasn't an entirely fair comparison. mpv has two DRM PRIME modes, one where it renders via EGL/VAAPI/Wayland, and one where it renders directly to an overlay plane. It tries the first before the second, but there isn't a way to explicitly select one of them at runtime. mpv tries to use desktop GL by default. This was failing under the Pi 3, so it fell back to the second mode. Desktop GL did work under the Pi 4, so it used the first mode. Telling it to use GLES instead made it slightly faster but not much. Eventually, I patched it to force the second mode. Now it works fine. The fact that it's slower isn't surprising. No GL filtering is done on the second mode, and you can make the first mode sufficiently performant by choosing bilinear filtering rather than lanczos. I think you should still be able to choose though, so I'll contact the mpv developers about this.

File: media-video/mpv/mpv-9999.ebuild [hunk 1]
```diff
@@ -44,7 +44,6 @@ REQUIRED_USE="
 "
 RESTRICT="!test? ( test )"
 
-# raspberry-pi: default to -bin given non-bin is known broken (bug #893422)
```
Comment (ionenwks) [inline media-video/mpv/mpv-9999.ebuild:47]: Can do `Closes: https://bugs.gentoo.org/893422`, glad won't have to worry about that messy package anymore.

Beside that, lgtm for mpv. Don't have a rpi so I'll trust whatever you feel works out better for it, thanks.

I'll keep myself a note to cleanup the mpv rpi profile stuff after stable+cleanup.

Comment (ionenwks) [reply media-video/mpv/mpv-9999.ebuild:47]: On a side-note, mpv-0.37.0 has new defaults that may be hard on a rpi compared to 0.36.0

If have speed issues, try --profile=fast to revert back to mpv-0.36.0 defaults.

Comment (chewi) [reply media-video/mpv/mpv-9999.ebuild:47]: I didn't realise that changed in 0.37.0. I am aware of `--profile=fast`, which I gather is basically the same as `--scaling=bilinear`. That's fine, but it still might be nice to be able to choose rendering directly to the overlay.

**Comment by chewi** [comment]
My plan to enable the `soc` USE flag by default for arm(64) and riscv worked better when I had patches against the latest ffmpeg. That's less feasible with John's patches. Simply enabling the flag won't pin ffmpeg back. Even forcing it wouldn't do that. I don't want to mask newer versions on those arches. I had hoped that putting `media-video/ffmpeg[soc]` in the world file would help but no. Strange though, as I think this does work in Portage sets. Better ideas welcome.

**Comment by chewi** [comment]
Turns out mpv _does_ have an option to render directly to the plane, it's `--gpu-hwdec-interop=drmprime-overlay`.

**Comment by chewi** [comment]
I've merged raspberrypi-utils and bumped rpi-eeprom already. John also released his patchset for ffmpeg 6.1, so I've updated accordingly. Not tested that yet, but it does build.

File: media-video/mpv/mpv-9999.ebuild [hunk 2]
```diff
@@ -216,7 +208,6 @@ src_configure() {
 		$(meson_feature libmpv plain-gl)
 		$(mpv_feature_multi opengl X gl-x11)
 		$(mpv_feature_multi opengl aqua gl-cocoa)
-		$(meson_feature raspberry-pi rpi)
```
Comment (ionenwks) [inline media-video/mpv/mpv-9999.ebuild:219]: I merged the changes for 9999 given this happened: https://github.com/mpv-player/mpv/commit/343a5fd345a84572ea5693a805119067d5b0fbc8

**Comment by chewi** [comment]
> I had hoped that putting `media-video/ffmpeg[soc]` in the world file would help but no. Strange though, as I think this does work in Portage sets. Better ideas welcome.

Turns out this does work. Subsequently doing `emerge ffmpeg` won't enable the flag, but `emerge @world` will.

Rebasing this over and over is getting tiring, so I'm just going to go ahead and merge it without libcamera this weekend. It's not like I'm masking or dropping anything yet.

**Comment by chewi** [comment]
Merged!

**Pull request closed without merging**